packages/core/src/TxMetadata/cip20.ts

```diff
@@ -0,0 +1,56 @@
+import { StringUtils } from '@cardano-sdk/util';
+import { TxMetadata } from '../Cardano';
+
+export const CIP_20_METADATUM_LABEL = 674n;
+
+export enum MessageValidationFailure {
+  wrongType = 'WrongType',
+  oversize = 'Oversize'
+}
+
+export type MessageValidationResult = {
+  valid: boolean;
+  failure?: MessageValidationFailure;
+};
+
+export type CIP20TxMetadataMessage = string;
+
+export type CIP20TxMetadataArgs = {
+  // An array of message strings, limited to 64 bytes each
+  messages: CIP20TxMetadataMessage[];
+};
+
+export type ValidationFailureMap = Map<CIP20TxMetadataMessage, MessageValidationResult>;
+
+/**
+ * Validate each message for correct type, in the case of JavaScript, and size constraint
+ *
+ * @param entry unknown
+ * @returns Validation result
+ */
+export const validateMessage = (entry: unknown): MessageValidationResult => {
+  if (typeof entry !== 'string') return { failure: MessageValidationFailure.wrongType, valid: false };
+  if (StringUtils.byteSize(entry) > 64) return { failure: MessageValidationFailure.oversize, valid: false };
+  return { valid: true };
+};
+
+export const isTxMetadata = (
+  value: TxMetadata | Map<CIP20TxMetadataMessage, MessageValidationResult>
+): value is TxMetadata => value.has(CIP_20_METADATUM_LABEL as never);
+
+/**
+ * Converts an object containing an array of individual messages into https://cips.cardano.org/cip/CIP-20 compliant
+ * transaction metadata
+ *
+ * @param args CIP20TxMetadataArgs
+ * @returns Transaction metadata or validation failures
+ */
+export const toCIP20Metadata = (args: CIP20TxMetadataArgs): TxMetadata | ValidationFailureMap => {
```

Comment:
This is a usability issue, as both the result and the error are a Map - difficult to differentiate between the 2
```suggestion
export const toCIP20Metadata = (args: CIP20TxMetadataArgs): Result<TxMetadata, ValidationFailureMap> => {
```

Reply:
Alternative approach in https://github.com/input-output-hk/cardano-js-sdk/pull/1427/commits/af0c3f3a102259726ea5f541d3e57d6d6272c35c